test/subshell/test_runner_test.go

```diff
@@ -34,16 +34,20 @@ func TestMockingRunner(t *testing.T) {
 
 	t.Run("MockCommitMessage", func(t *testing.T) {
 		t.Parallel()
+		dir := t.TempDir()
 		runner := subshell.TestRunner{
-			WorkingDir: t.TempDir(),
-			HomeDir:    t.TempDir(),
-			BinDir:     filepath.Join(t.TempDir(), "bin"),
+			WorkingDir: dir,
+			HomeDir:    dir,
+			BinDir:     filepath.Join(dir, "bin"),
 		}
 
 		runner.MockCommitMessage("test commit message")
 
-		res := runner.MustQuery("bash", "-c", "$GIT_EDITOR output; cat output")
-		must.Eq(t, "test commit message", res)
+		// Verify that `$GIT_EDITOR file` sets the file's contents to the given commit message.
```

Comment:
This is a pretty clever test because it bypasses Git altogether! :+1: A comment like this would help people like me follow what it does:

```suggestion
		// Simulate Git calling the executable configured in the $GIT_EDITOR environment variable when it needs to display an editor to the user.
```

Reply:
I propose a slightly more verbose comment, please take another look.